lib/client.js

```diff
@@ -326,6 +326,7 @@ export default class Client {
     this._source = options.source || (this._parent && this._parent._source) || window.parent
     this._appGuid = options.appGuid || (this._parent && this._parent._appGuid)
     this._instanceGuid = options.instanceGuid || this._appGuid
+    this._allowOrigin = options.allowOrigin || false
```

Comment:
I suspect that this might not work, because in practice it is the app that initialises the sdk_client [here](https://github.com/zendesk/zendesk_app_framework_sdk/blob/master/lib/index.js#L38).  You would likely need to introduce that as an additional option to be passed either explicitly or implicitly in index.js.

Of course if you did end up implementing this you would be bypassing the whole reason that https://github.com/zendesk/zendesk_app_framework_sdk/pull/58 was done, which was to implement a client-side security feature for the zaf_sdk.  We evidently can't do this, so there would be a need to implement this another way.

I gather that the usecase here is end-user apps, which is moderately interesting.

I gather you're somehow interested in allowing zaf_sdk to pick up a message from our systems and for this you need to work around the allowed origins whitelist.

One potential way to do this might be to write the desired message for zaf_sdk to a pub/sub system eg Kafka or SQS, and then we could have logic in zaf_sdk that polls a whitelisted endpoint in a separate VPC to our other infrastructure (eg like an Untrusted Ingress Zone).  Then somehow we'd like to prove that the request is coming from a legitimate end user app, it seems a bit unclear how to do that ... but assuming that we do have a way to do that, then this nascent ingress service could direct the request to a consumer that listens to the topic, and then the request could be returned to the app via the existing egress lambda service.

Probably for proving that the request is legitimate potentially we could adapt the pattern used in https://github.com/zendesk/zendesk_app_framework_sdk/pull/145 to have a state associated to the app that is a string stored in an object associated to the zaf_sdk, and this is coupled to classic which generates a 1 time password string every `<interval>` eg every 30 seconds, and this could be kept in sync with the consumer in the separate VPC that I've indicated above.

Or we could adopt another approach that is a bit less convoluted but still secure.

But we unfortunately cannot solve your problem via this approach, we'd need to likely adopt a slightly more complex solution.